Add WordPress-Gatsby

henrikwirth opened the pull request:
https://github.com/henrikwirth/awesome-wordpress-gatsby

A curated list of resources about WordPress as a headless CMS with Gatsby


PRs reviewed: 
#1622 
#1623 

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
- You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The suggested Awesome list complies with the below requirements.


## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- **If you use a license badge, it should be SVG, not PNG.**
	- To verify that you've read all the guidelines, please add comment with just the word `unicorn`.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

caufieldjh commented:
Descriptions of resources in the list would be very helpful.
Might not be a bad idea to read over the requirements list again.

henrikwirth commented:
@caufieldjh 

> Needs a period at the end of the list description

This is done, for the one description I had.

> Descriptions of resources in the list would be very helpful.

This I guess makes sense for all the Technology-Based links. I feel the Tutorials are mostly self-explanatory. What do you think?

> Might not be a bad idea to read over the requirements list again.

Anything else missing besides the descriptions?

jhngrant commented:
In the Starters & Themes section, provide a brief description of what a Starter and Theme is. This will both signpost beginners and help the project gain traction.

At the beginning of the README include a simple explanation of what headless CMS is and/or the benefits of this approach/architecture.

caufieldjh commented on readme.md:
```diff
@@ -257,6 +257,7 @@
 - [Hyperapp](https://github.com/jorgebucaran/awesome-hyperapp#readme) - Tiny JavaScript library for building web apps.
 - [lit-html](https://github.com/web-padawan/awesome-lit-html#readme) - HTML templating library for JavaScript.
 - [JAMstack](https://github.com/automata/awesome-jamstack#readme) - Modern web development architecture based on client-side JavaScript, reusable APIs, and prebuilt markup.
+- [WordPress-Gatsby](https://github.com/henrikwirth/awesome-wordpress-gatsby#readme) - Web development technology stack with WordPress as a backend and Gatsby as a frontend
```
> This is done, for the one description I had.
I mean this one.

henrikwirth replied:
Done, thanks.

henrikwirth commented:
> In the Starters & Themes section, provide a brief description of what a Starter and Theme is. This will both signpost beginners and help the project gain traction.
> 
> At the beginning of the README include a simple explanation of what headless CMS is and/or the benefits of this approach/architecture.

I added the descriptions for themes, starters and about static-site-generators + headless CMS. Thanks for info.

henrikwirth commented:
> Descriptions of resources in the list would be very helpful.

I added descriptions to all of the resources. Thanks for mentioning.

sindresorhus commented on readme.md:
```diff
@@ -257,6 +257,7 @@
 - [Hyperapp](https://github.com/jorgebucaran/awesome-hyperapp#readme) - Tiny JavaScript library for building web apps.
 - [lit-html](https://github.com/web-padawan/awesome-lit-html#readme) - HTML templating library for JavaScript.
 - [JAMstack](https://github.com/automata/awesome-jamstack#readme) - Modern web development architecture based on client-side JavaScript, reusable APIs, and prebuilt markup.
+- [WordPress-Gatsby](https://github.com/henrikwirth/awesome-wordpress-gatsby#readme) - Web development technology stack with WordPress as a backend and Gatsby as a frontend.
```
```suggestion
- [WordPress-Gatsby](https://github.com/henrikwirth/awesome-wordpress-gatsby#readme) - Web development technology stack with WordPress as a back end and Gatsby as a front end.
```

henrikwirth replied:
Good catch

henrikwirth replied:
@sindresorhus wouldn't it make sense to use front-end and back-end instead? Also in the Readme are two approaches `Frontend development` under `Platforms` and `Front-End Development` as a topic.

sindresorhus commented:
> Consistent and correct naming. For example, Node.js, not NodeJS or node.js.

GatsbyJS => Gatsby

sindresorhus commented:
I think the readme intro could be clearer about the benefit of combining WordPress and Gatsby.

henrikwirth commented:
> > Consistent and correct naming. For example, Node.js, not NodeJS or node.js.
> 
> GatsbyJS => Gatsby

Wasn't sure how to name it, as GatsbyJS is using different approaches on their site. But I guess GatsbyJS makes it a bit more clear. I'll update everything accordingly.

> I think the readme intro (in your repo) could be clearer about the benefit of combining WordPress and Gatsby.

I'll add a paragraph about it, under the description of headless CMS and static site generators. Wouldn't explain it too much though, as I think it is very biased and the pros and cons are depending on a lot of things. I would rather add some resources, who talk about it, so everyone can figure out themselves, if it is the right approach for their usecase.

henrikwirth commented:
@sindresorhus I changed everything to Gatsby now instead of GatsbyJS and added the paragraph about the benefit of the approach. Also committed your suggested change.

Now lets hope this list has all the awesomeness it needs now :)

sindresorhus commented:
Looks good now :)

sindresorhus commented:
Tweet: https://twitter.com/awesome__re/status/1194888938986430465
